syntax = "proto2";

package st.planning;
// option go_package = "st.proto.ai/onboard/planning.common/proto/planning.status";

message PlannerStatusProto {
  enum PlannerStatusCode {
    OK = 0;
    FAIL = 1;
    // input_topic 2~29
    INPUT_INCORRECT = 2;
    PLAN_MSG_MAP_TIMEOUT = 3;
    PLAN_FAIL_NO_MAP = 4;
    PLAN_MSG_OBSTACLE_TIMEOUT = 5;
    PLAN_MSG_OBSTACLE_DELAY = 6;
    PLAN_MSG_ODOMETRY_TIMEOUT = 7;
    PLAN_MSG_ODOMETRY_DELAY = 8;
    PLAN_MSG_VEHICLE_TIMEOUT = 9;
    PLAN_MSG_VMC_TIMEOUT = 10;
    PLAN_MSG_BEHAVIOR_TIMEOUT = 11;
    OBJECT_MANAGER_FAILED = 12;
    FALLBACK_PREVIOUS_INFO_UNAVAILABLE = 13;
    START_POINT_PROJECTION_TO_ROUTE_FAILED = 14;
    TARGET_LANE_JUMPED_FAIL = 15;
    EGO_LANE_LOSS_FAIL = 16;
    PLAN_MSG_MAP_EVENT_TIMEOUT = 17;
    // Scheduler 30~49
    SCHEDULER_UNAVAILABLE = 30;
    TARGET_LANE_CANDIDATES_UNAVAILABLE = 31;
    // Decision 50~69
    DECISION_CONSTRAINTS_UNAVAILABLE = 50;
    // Initializer 70~89
    INITIALIZER_FAILED = 70;
    LC_LINE_CHECK_FAILED = 71;
    LC_MISS_NAVI_FAILED = 72;
    LC_SPECIAL_LANE_CHECK_FAILED = 73;
    TRAJECTORY_NOT_ENTERING_TARGET_LANE = 74;
    FRONT_DANGEROUS_VEHICLE_TARGET_LANE = 75;
    REAR_DANGEROUS_VEHICLE_TARGET_LANE = 76;
    FRONT_DANGEROUS_VEHICLE_CURRENT_LANE = 77;
    REAR_DANGEROUS_VEHICLE_CURRENT_LANE = 78;
    DANGEROUS_STATIONARY_OBSTACLE = 79;
    // Optimizer 90~109
    OPTIMIZER_FAILED = 90;
    PATH_EXTENSION_FAILED = 91;
    TRAJECTORY_VALIDATION_FAILED = 92;
    LC_SAFETY_CHECK_FAILED = 93;
    PLANNER_ABNORMAL_EXIT = 94;
    // Speed 110~129
    SPEED_OPTIMIZER_FAILED = 110;
    // Selector 130~149
    SELECTOR_FAILED = 130;
    BRANCH_RESULT_IGNORED = 131;
    DEVIATE_NAVI = 132;
    // Acc 150~169
    ACC_CORRIDOR_FAILED = 150;
    ACC_SPEED_FAILED = 151;
    ACC_TRAJECTORY_VALIDATION_FAILED = 152;
  }
  optional PlannerStatusCode status = 1;
  optional string message = 2;
}
